dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/Cost.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.core.metrics;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+@Schema(name = "Cost", description = "Cost represents a monetary amount in USD")
+public class Cost {
+    private static final String DEFAULT_CURRENCY = "USD";
+
+    // We may eventually want to allow different currencies to be submitted
+    @Schema(description = "The currency of the cost value", defaultValue = DEFAULT_CURRENCY)
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    private String currency = DEFAULT_CURRENCY;
+
+    @Schema(description = "The numerical value of the cost", example = "5.99")
+    private Double value;
```

Comment:
Is there any reason this is `Double` rather than `double`?
Do we want to support a null cost intentionally?

Reply:
Good catch. Since there's been discussion of the use of `double` vs `BigDecimal`, I think `BigDecimal` might be better here instead

Reply:
Yeah, my superficial reading of things to be super-safe is just to keep the initial values around as `String`, construct `BigDecimal` or `Money` from them directly and never use `double`

I suspect things like mins and maxes are ok for the precision we're dealing with, but it may be simpler for our mental load to just use the above rule of thumb.

Reply:
> Yeah, my superficial reading of things to be super-safe is just to keep the initial values around as String, construct BigDecimal or Money from them directly and never use double

My hesitation about using strings is that we would need to validate that the provided string is a valid BigDecimal so I prefer to skip that extra validation and just make it a BigDecimal

Reply:
See https://github.com/dockstore/dockstore/pull/5577#discussion_r1258345051